api/client/trust.go

```diff
@@ -0,0 +1,534 @@
+	}
+	fs.BoolVar(&untrusted, []string{"-untrusted"}, !trusted, message)
+}
+
+func isTrusted(name string) bool {
+	// TODO(dmcgowan): false if name is hex image id
+	// TODO(dmcgowan): whitelist specific names to be untrusted
+	// TODO(dmcgowan): canonicalize name before checking whitelist
+	return !untrusted
+}
+
+var targetRegexp = regexp.MustCompile(`([\S]+): digest: ([\S]+) size: ([\d]+)`)
+
+// dropCR drops a terminal \r from the data.
+func dropCR(data []byte) []byte {
+	if len(data) > 0 && data[len(data)-1] == '\r' {
+		data = data[0 : len(data)-1]
+	}
+	if len(data) > 0 && data[0] == '\r' {
+		data = data[1:]
+	}
+	return data
+}
+
+// escapeSequenceLength calculates the length of an ANSI escape sequence
+// If there is not enough characters to match a sequence, -1 is returned,
+// if there is no valid sequence 0 is returned, otherwise the number
+// of bytes in the sequence is returned.
+func escapeSequenceLength(data []byte) int {
+	next := 0
+	if len(data) <= next {
+		return -1
+	}
+	if data[next] != '[' {
+		return 0
+	}
+	for {
+		next = next + 1
+		if len(data) <= next {
+			return -1
+		}
+		if data[next] > '9' || data[next] < '0' || data[next] == ';' {
+			break
+		}
+	}
+	if len(data) <= next {
+		return -1
+	}
+	code := data[next]
+	if (code >= 'A' && code <= 'K') || code == 'm' {
+		return next + 1
+	}
+
+	return 0
+}
+
+func ScanInput(data []byte, atEOF bool) (advance int, token []byte, err error) {
+	if atEOF && len(data) == 0 {
+		return 0, nil, nil
+	}
+
+	// Look for escape sequence
+	if i := bytes.IndexByte(data, '\x1b'); i >= 0 {
+		last := i + 1
+		// get length of ANSI escape sequence
+		if sl := escapeSequenceLength(data[i+1:]); sl != -1 {
+			last = last + sl
+		} else {
+			return 0, nil, nil
+		}
+
+		return last, dropCR(data[0:i]), nil
+	}
+	if i := bytes.IndexByte(data, '\n'); i >= 0 {
+		// No escape sequence, check for new line
+		return i + 1, dropCR(data[0:i]), nil
+	}
+
+	// If we're at EOF, we have a final, non-terminated line. Return it.
+	if atEOF {
+		return len(data), dropCR(data), nil
+	}
+	// Request more data.
+	return 0, nil, nil
+}
+
+type target struct {
+	reference registry.Reference
+	digest    digest.Digest
+	size      int64
+}
+
+func (cli *DockerCli) trustDirectory() string {
+	return filepath.Join(cliconfig.ConfigDir(), "trust")
+}
+
+// certificateDirectory returns the directory containing
+// TLS certificates for the given server. An error is
+// returned if there was an error parsing the server string.
+func (cli *DockerCli) certificateDirectory(server string) (string, error) {
+	u, err := url.Parse(server)
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(cliconfig.ConfigDir(), "tls", u.Host), nil
+}
+
+func trustServer(index *registry.IndexInfo) string {
+	if s := os.Getenv("NOTARY_SERVER"); s != "" {
+		if !strings.HasPrefix(s, "http") {
+			return "https://" + s
+		}
+		return s
+	}
+	if index.Official {
+		return "https://notary-stage.docker.io"
```

Comment:
I take it these images have not been pushed and/or the infrastructure is not in place yet, is there an estimated date for this. I really am not a fan of going down this road it is super stressful for all when the stuff is not in place a day before the release and it has not been properly tested.